packages/workspace/src/workspace/nacl_files/multi_env/routers.ts

```diff
@@ -49,7 +49,10 @@ const separateChangeByFiles = async (
   (await source.getSourceRanges(change.id))
     .map(range => range.filename)
     .map(async filename => {
-      const pathHint = _.trimEnd(filename, FILE_EXTENSION).split(path.sep)
+      const parsedPath = path.parse(filename)
+      const pathHint = (parsedPath.dir
+        ? [...parsedPath.dir.split(path.sep), parsedPath.name]
+        : [parsedPath.name])
```

Comment:
a couple of slightly simpler options
1. since we know the filename extension is going to be `.nacl` we could just
```suggestion
      const pathHint = filename.slice(0, -FILE_EXTENSION.length).split(path.sep)
```
2. if we don't want to rely on that and we just want to get rid of the file extension, we could
```suggestion
      const pathHint = filename.slice(0, -path.extname(filename).length).split(path.sep)
```

Reply:
I didn't want to assume the file extension and if there are any filenames with no extension, suggestion 2 would break (and if we add handling for that it's no longer going to be simpler than the current one). Do you think it's better than just keeping the extra line? I could make it a one-liner but I think it's more readable this way.

Reply:
the reason I suggested these options are that it really isn't very clear what the new code is trying to do.
with the old code the intention to remove the suffix and split by path sep was very clear. with the new code it is really hard to figure out what is supposed to happen and why it is there.

Reply:
trying another variant, wdyt?